src/ui/public/timepicker/timepicker.js

```diff
@@ -76,11 +76,19 @@ module.directive('kbnTimepicker', function (quickRanges, timeUnits, refreshInter
       });
 
       $scope.$watch('absolute.from', function (date) {
-        if (_.isDate(date)) $scope.absolute.from = moment(date);
+        // If it is a date, it was set from the timepicker, so we set it to the start of day and transform it into a moment
+        if (_.isDate(date)) {
+          date.setHours(0, 0, 0, 0); // Start of day
```

Comment:
Does it need to do anything special to support it? I thought the getterSetter would be declared in timepicker.js and it would work without any modification to datepicker, since it would look like a regular model from datepicker's point of view.